### PR TITLE
add return types and unify subscribing to listeners

### PR DESCRIPTION
**Pull request by fpkamp**

### CODE REVIEW

File: node/src/browserstack_reporter.ts [hunk 1]
```diff
@@ -14,10 +14,10 @@ export function BrowserStackReporter(
   const log = logger.create('Browserstack Reporter')
 
   let pendingUpdates = 0
-  // eslint-disable-next-line @typescript-eslint/no-empty-function
-  let callWhenFinished = () => {}
 
-  const exitIfAllFinished = function () {
+  let callWhenFinished = () => undefined
```
[the remark this answers is not in the document]
Comment (fpkamp) [reply node/src/browserstack_reporter.ts:18]:
```
src/browserstack_reporter.ts:20:3 - error TS1109: Expression expected.

20   const exitIfAllFinished = () => {
     ~~~~~
```

[the remark this answers is not in the document]
Comment (Finesse) [reply node/src/browserstack_reporter.ts:18]: Show the code. This is correct:

image

[the remark this answers is not in the document]
Comment (fpkamp) [reply node/src/browserstack_reporter.ts]: saw your comment before you edited it (`() => void`), now it's okay

[the remark this answers is not in the document]
Comment (Finesse) [reply node/src/browserstack_reporter.ts]: My bad, sorry

**Pull request merged**